pages/settings/BitcoinMap.tsx

```diff
@@ -0,0 +1,53 @@
+import * as Location from "expo-location";
+import React from "react";
+import { View } from "react-native";
+import { WebView } from "react-native-webview";
+import Loading from "~/components/Loading";
+import Screen from "~/components/Screen";
+
+import { Text } from "~/components/ui/text";
+import { errorToast } from "~/lib/errorToast";
+
+export function BitcoinMap() {
+  const [isLoading, setIsLoading] = React.useState(true);
+  const [mapUrl, setMapUrl] = React.useState("https://btcmap.org/map");
+
+  React.useEffect(() => {
+    (async () => {
+      try {
+        const { status } = await Location.requestForegroundPermissionsAsync();
+        if (status !== Location.PermissionStatus.GRANTED) {
+          throw new Error("Permission to access location was denied.");
+        }
+
+        const location = await Location.getCurrentPositionAsync({});
+        setMapUrl(
+          `https://btcmap.org/map#18/${location.coords.latitude}/${location.coords.longitude}`,
+        );
+      } catch (error) {
+        errorToast(error);
+      } finally {
+        setIsLoading(false);
+      }
+    })();
+  }, []);
+
+  return (
+    <View className="flex-1">
+      <Screen title="Bitcoin Map" />
+
+      {isLoading ? (
+        <View className="flex-1 justify-center items-center">
+          <Loading />
+          <Text className="mt-4">Loading BTC Map</Text>
+        </View>
+      ) : (
+        <WebView
+          source={{ uri: mapUrl }}
+          className="flex-1 bg-blue-100"
+          geolocationEnabled
```

Comment:
should this be based on if the permission was granted?

Reply:
why do we need that after all? don't we set the location from RN by updating the location to the coordinates?

Reply:
Yeah, but we need the location again for Webview separately if you click on the current position icon in `BTC Map` to get the current live location

Also got to know that this is for Android only, for iOS we have to do [this](https://stackoverflow.com/questions/74347489/how-to-pass-geolocation-permission-to-react-native-webview), is that okay?

> should this be based on if the permission was granted?

Will do that once we get the green flag for above (or we remove entirely so the user can give access again (given they clicked on position icon, otherwise BTCMap doesn't need it))

Reply:
> Also got to know that this is for Android only, for iOS we have to do this, is that okay?

I'd argue this code is not worth adding. If the map is already zoomed at your current location, who will use this button in the map anyway? 

> (or we remove entirely so the user can give access again (given they clicked on position icon, otherwise BTCMap doesn't need it))

What do you mean with "remove entirely"?